lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -118,6 +119,53 @@ static Value createTensorSub(PatternRewriter &rewriter, Location loc,
   return sub;
 }
 
+static Value getDtypeIntValueForType(PatternRewriter &rewriter, Location loc,
+                                     Type dtype) {
+  int intType = (int)getScalarTypeForType(dtype);
+  return rewriter.create<ConstantIntOp>(loc,
+                                        rewriter.getI64IntegerAttr(intType));
+}
+
+// Helper to convert a tensor to a specific scalar type.
+static Value convertTensorToDtype(PatternRewriter &rewriter, Location loc,
+                                  Value input, Type dtype) {
+  BaseTensorType origType = input.getType().cast<BaseTensorType>();
+  Type newType = origType.getWithSizesAndDtype(origType.getSizes(), dtype);
+  // `intType` contains the corresponding integer for the dtype which is used
+  // by the aten.to.dtype op.
+  Value convertIntVal = getDtypeIntValueForType(rewriter, loc, dtype);
+  Value falseVal = rewriter.create<ConstantBoolOp>(loc, false);
+  Value noneVal = rewriter.create<ConstantNoneOp>(loc);
+  Value converted = rewriter.create<AtenToDtypeOp>(
+      loc, newType, input, convertIntVal, falseVal, falseVal, noneVal);
+  return converted;
+}
+
+// Helper to create a tensor filled with the given scalar.
+static Value createInitTensor(PatternRewriter &rewriter, Location loc,
+                              Type resultType, Value scalar, Value sizeList) {
+  BaseTensorType tensorType = resultType.cast<BaseTensorType>();
+  Value noneVal = rewriter.create<ConstantNoneOp>(loc);
+  Value emptyTensor = rewriter.create<AtenEmptyMemoryFormatOp>(
+      loc, tensorType, sizeList, /*dtype=*/noneVal, /*layout=*/noneVal,
+      /*device=*/noneVal,
+      /*pin_memory=*/noneVal, /*memory_format=*/noneVal);
+  return rewriter.create<PseudoAtenFillScalarOp>(loc, resultType, emptyTensor,
+                                                 scalar);
+}
+
+// Helper to create a rank0 tensor filled with the given scalar.
+static Value createRank0Tensor(PatternRewriter &rewriter, Location loc,
+                               BaseTensorType inputType, Value scalar) {
+  SmallVector<int64_t> sizes;
+  Type rank0TensorTy = inputType.getWithSizesAndDtype(
```

Comment:
In case the scalar is float and the input type is integer, the result will be truncated. Can you please update the docs wrt that to avoid any misuses of this helper.